src/SFML/Graphics/RenderTarget.cpp

```diff
@@ -663,6 +739,41 @@ void RenderTarget::applyBlendMode(const BlendMode& mode)
 }
 
 
+////////////////////////////////////////////////////////////
+void RenderTarget::applyStencilMode(const StencilMode& mode)
+{
+    using RenderTargetImpl::stencilFunctionToGlConstant;
+    using RenderTargetImpl::stencilOperationToGlConstant;
+
+    // Fast path if we have a default (disabled) stencil mode
+    if (mode == StencilMode())
```

Comment:
The default constructed `StencilMode` should correspond to the default state of stencil operations in a newly created OpenGL context. In this case, rendering is performed as if no stencil testing is required. Because there is always a stencil state (even if it effectively doesn't do anything) in OpenGL, using an `optional` to model this wouldn't make sense.